site2/docs/cookbooks-message-throttling.md

```diff
@@ -0,0 +1,86 @@
+  throttling makes Pulsar work steadily by keeping Storage's read request load stable.
+* An instance of 'Broker' serves multiple topics at the same time, and if a topic is too busy, it will occupy
+  almost all of the IO resources, other topics will not work well. Use message dispatch throttling can balance
+  the allocation of resources to agents across topics.
+* There have large backlog of messages to consume, clients may receive a large amount of data in a short period of time,
+  thus monopolizing the client computer resources. Message dispatch throttling can also be used to balance the resource
+  allocation of the client computer.
+
+### Concepts of message dispatch throttling
+- `ratePeriodInSecond` The unit of the rate limiting period is second. The default is 1s.
+- `dispatchThrottlingRateInMsg` Specifies the maximum number of messages to be delivered in each rate limiting period. 
+   The default is' -1 ', which means no limit.
+- `dispatchThrottlingRateInByte` The maximum number of bytes of messages delivered per rate-limiting period. The default
+   is' -1 ', which means no limit.
+
+> If set ` dispatchThrottlingRateInMsg` and ` dispatchThrottlingRateInByte` both, then the message delivery is required 
+> at the same time satisfy the rules of the two, in other words the two rules are not mutually exclusive.
+
+### How it works
+Message dispatch throttling works divided into these steps:
+1. Calculate the number of messages or bytes to be delivered.
+2. Estimate the amount of data to be read from the Storage (This estimate logic is not accurate, as described below).
+3. Update the counter of message dispatch throttling.
+4. Actually deliver the message to the client.
+
+> If the quota in the current rate limiting period is not used up, the quota will not be used in the next rate limiting
+> period. However, if the quota in the current rate limiting period is exceeded, the quota in the next rate limiting
+> period is reduced. For example, if the rate-limiting rule is set to `10 /s`, `11` messages are delivered to the client
+> in the first rate-limiting period,`9` messages will be delivered to the client in the next rate-limiting period.
+
+#### Estimate the amount of data to be read from the Storage
+Messages is stored in blocks. If batch-send is not enabled, each message is packaged into an independent data block.And
+if batch-send is enabled, a batch of messages are packaged into one data block, then we can not determine the true count
+of messages in one data block; We also cannot determine the true size of this data block until we have actually read the
+it from Storage.So in order to satisfy the constraints `dispatchThrottlingRateInMsg` and `dispatchThrottlingRateInByte`,
+requires a mechanism to estimate how much need to read a data block.
+
+> If enabled the feature ` dispatchThrottlingOnBatchMessageEnabled`, instead of counting by message, we're counting
```

Comment:
> Could we explain more about this feature dispatchThrottlingOnBatchMessageEnabled? 

already add more description about `dispatchThrottlingOnBatchMessageEnabled`.

> If this is enabled, are dispatchThrottlingRateInMsg and dispatchThrottlingRateInByte set for each batch instead of individual messages?

no, only `dispatchThrottlingRateInMsg`